flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,540 @@
+                            if self.config.get('type') == 'seasons':
+                                entry = self.generate_tv_entry(
+                                    parent_request, child_request, season
+                                )
+                                if entry:
+                                    self._items.append(entry)
+                            else:
+                                for episode in season['episodes']:
+                                    # check that the request is approved unless user has selected to include everything
+                                    if (
+                                        self.config.get('only_approved')
+                                        and not episode.get('approved')
+                                        or episode.get('approved')
+                                    ):
+                                        # Always include items that are not available and only include available items if user has selected to do so
+                                        if (
+                                            self.config.get('include_available')
+                                            and episode.get('available')
+                                        ) or not episode.get('available'):
+                                            entry = self.generate_tv_entry(
+                                                parent_request, child_request, season, episode
+                                            )
+                                            self._items.append(entry)
+        return self._items
+
+    @property
+    def online(self):
+        """Set the online status of the plugin, online plugin should be treated differently in certain situations,
+        like test mode"""
+        return True
+
+    # -- Public interface ends here -- #
+
+    def _get_ombi_entry(self, entry: Entry) -> Optional[OmbiEntry]:
+
+        if "tmdb_id" not in entry:
+            log.warning(
+                f"{entry['title']} is missing the tmdb_id, consider using tmdb_lookup plugin."
+            )
+
+        entry_type: str = self.config['type']
+
+        if entry_type == 'movies':
+            return OmbiMovie.from_imdb_id(self._get_base_url(), entry['tmdb_id'], self.ombi_auth)
+
+        return OmbiTv.from_tvdb_id(self._get_base_url(), entry['tmdb_id'], self.ombi_auth)
+
+    def generate_series_id(self, season, episode=None):
+        tempid = 'S' + str(season.get('seasonNumber')).zfill(2)
+        if episode:
+            tempid = tempid + 'E' + str(episode.get('episodeNumber')).zfill(2)
+
+        return tempid
+
+    def generate_title(self, item, season=None, episode=None):
+        temptitle = item.get('title')
+
+        if item.get('releaseDate') and self.config.get('include_year'):
+            temptitle = temptitle + ' (' + str(item.get('releaseDate')[0:4]) + ')'
```

Comment:
Questionable str cast and list slicing, if release date is None it will produce "Sometitle (None)". It would be better not to try putting anything in parens without releaseDate

Reply:
Agreed, but it wasn't a problem the last couple of years. Lets fix it when its a problem 😊